ext/PathfinderTuringExt.jl

```diff
@@ -110,10 +164,13 @@ function Pathfinder.pathfinder(
     kwargs...,
 )
     var_names = flattened_varnames_list(model)
-    prob = Turing.optim_problem(model, Turing.MAP(); constrained=false, init_theta=init)
-    init_sampler(rng, prob.prob.u0)
-    result = Pathfinder.pathfinder(prob.prob; rng, input=model, kwargs...)
-    draws = reduce(vcat, transpose.(prob.transform.(eachcol(result.draws))))
+    # If no initial value is provided, sample from prior.
+    init = init === nothing ? rand(Vector, model) : init
+    f, init, transform_result = set_up_model_optimisation(model, init)
+    prob = Optimization.OptimizationProblem(f, init)
+    init_sampler(rng, init)
```

Comment:
Since the `init` doesn't seem to be necessary for `set_up_model_optimisation`, we can just wait to handle it until after it's set up.
```suggestion
    f, init_point, transform_result = set_up_model_optimisation(model)
    prob = Optimization.OptimizationProblem(f, init_point)
    if init === nothing
        init_sampler(rng, init_point)
    else
        copyto!(init_point, init)
    end
```

Reply:
If the `init` provided by the caller is in the constrained space of the original parameters, wouldn't this result in calling the optimiser with an initial value in the constrained space even though the optimisation happens in the constrained one? The idea being that passing `init` through `set_up_model_optimisation` transforms it to the unconstrained space.

Reply:
Ah, currently we only support unconstrained `init`s. I can in a later breaking PR work on supporting constrained ones.

Reply:
Oh I see, I had assumed the `init` was in constrained space. Indeed, all the changes you propose regarding getting rid of the `init` argument then make sense.